### ISSUE TITLE
[CLOSED] Simplify Arithmetic Priors

### ISSUE THREAD

**Issue by eriq-augustine** [linqs/psl#170]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [eriq-augustine](https://github.com/eriq-augustine)**
_Monday Nov 06, 2017 at 06:04 GMT_
_Originally opened as https://github.com/eriq-augustine/psl/issues/123_

----

Currently, all arithmetic rules using equality grounds two instances: `<=` and `>=`.
However, logical priors (logical rules with one atom) only ground one instance.
We should detect when an arithmetic rule only has one atom and make sure we only ground one instance.
(Not that this only applies for non-summation rules.)

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Monday Jul 23, 2018 at 02:12 GMT_

----

I think we need to abstract this one more level.

It is not so much a problem that two ground rule are generated for something like:
```
1.0: Foo(A, B) = 0 ^2
```

But rather that we get two rules like:
```
1.0: Foo('Alice', 'Bob') <= 0.0 ^2
1.0: Foo('Alice', 'Bob') >= 0.0 ^2
```

Notice that the second rule is trivial, but only when the constant is 0.0.
(The first rule is trivial if the constant is 1.0.)

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Monday Jul 23, 2018 at 02:17 GMT_

----

Essentially, we need something the looks for trivial ground rules right before they are added to the GRS.

**Comment by eriq-augustine** [comment]
<a href="https://github.com/eriq-augustine"><img src="https://avatars0.githubusercontent.com/u/337857?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eriq-augustine](https://github.com/eriq-augustine)**
_Monday Jul 23, 2018 at 19:45 GMT_

----

Fixed in 42473c4ab43c98b71ee63cfbfb01c61f70e928f7.
